Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -335,6 +334,33 @@ public struct HTTPHeaders: CustomStringConvertible, ExpressibleByDictionaryLiter
         }
     }
 
+    /// Add a sequence of header name/value pairs to the block.
+    ///
+    /// This method is strictly additive: if there are other entries with the same header
+    /// name already in the block, this will add new entries.
+    ///
+    /// - Parameter contentsOf: The sequence of header name/value pairs. For maximum compatibility
+    ///     the header should be an ASCII string. For future-proofing with HTTP/2 lowercase header
+    ///     names are strongly recommended.
+    @inlinable
+    public mutating func add<S: Sequence>(contentsOf other: S) where S.Element == (String, String) {
+        precondition(!other.contains { !$0.0.utf8.contains(where: { !$0.isASCII }) }, "names must be ASCII")
+        self.headers.append(contentsOf: other)
```

Comment:
why doesn't this call `self.add` for every element (whilst reserving the capacity in `self.headers` beforehands)?

Reply:
Sure, makes sense!